web3/utils/abi.py

```diff
@@ -1,21 +1,686 @@
+            ...     "type": "function",
+            ... }
+            >>> check_if_arguments_can_be_encoded(abi, *[7, 3], **{})
+            True
+    """
+    if abi_element["type"] == "fallback" or abi_element["type"] == "receive":
+        return True
+
+    try:
+        arguments = get_normalized_abi_inputs(abi_element, *args, **kwargs)
+    except TypeError:
+        return False
+
+    if len(abi_element.get("inputs", ())) != len(arguments):
+        return False
+
+    try:
+        types, aligned_args = get_aligned_abi_inputs(abi_element, arguments)
+    except TypeError:
+        return False
+
+    if abi_codec is None:
+        abi_codec = ABICodec(default_registry)
+
+    return all(
+        abi_codec.is_encodable(_type, arg) for _type, arg in zip(types, aligned_args)
+    )
+
+
+def get_event_abi(
+    abi: ABI,
+    event_name: str,
+    argument_names: Optional[Sequence[str]] = None,
+) -> ABIEvent:
+    """
+    Find the event interface with the given name and/or arguments.
+
+    :param abi: Contract ABI.
+    :type abi: `ABI`
+    :param event_name: Find an event abi with matching event name.
+    :type event_name: `str`
+    :param argument_names: Find an event abi with matching arguments.
+    :type argument_names: `list[str]`
+    :return: ABI for the event interface.
+    :rtype: `ABIEvent`
+
+    .. doctest::
+
+        >>> from web3.utils import get_event_abi
+        >>> abi = [
+        ...   {"type": "function", "name": "myFunction", "inputs": [], "outputs": []},
+        ...   {"type": "function", "name": "myFunction2", "inputs": [], "outputs": []},
+        ...   {"type": "event", "name": "MyEvent", "inputs": []}
+        ... ]
+        >>> get_event_abi(abi, 'MyEvent')
+        {'type': 'event', 'name': 'MyEvent', 'inputs': []}
+    """
+    filters: List[functools.partial[Sequence[ABIElement]]] = [
+        functools.partial(filter_abi_by_type, "event"),
+    ]
+
+    if event_name is None or event_name == "":
+        raise Web3ValidationError(
+            "event_name is required in order to match an event ABI."
+        )
+
+    filters.append(functools.partial(filter_abi_by_name, event_name))
+
+    if argument_names is not None:
+        filters.append(functools.partial(filter_by_argument_name, argument_names))
+
+    event_abi_candidates: Sequence[ABIEvent] = pipe(abi, *filters)
+
+    if len(event_abi_candidates) == 1:
+        return event_abi_candidates[0]
+    elif len(event_abi_candidates) == 0:
+        raise Web3ValueError("No matching events found")
+    else:
+        raise Web3ValueError("Multiple events found")
+
+
+def get_event_log_topics(
+    event_abi: ABIEvent,
+    topics: Optional[Sequence[HexBytes]] = None,
+) -> Sequence[HexBytes]:
+    r"""
+    Return topics for an event ABI.
+
+    :param event_abi: Event ABI.
+    :type event_abi: `ABIEvent`
+    :param topics: Transaction topics from a `LogReceipt`.
+    :type topics: `list[HexBytes]`
```

Comment:
Should this be optional? Is there a reason that it is? imo this should be a required arg and not an optional kwarg unless there's utility in keeping this optional. If the latter, then this doctstring should change to:

```suggestion
    :type topics: `Optional[Sequence[HexBytes]]`
```

Reply:
That makes sense, I'll update!